### PR TITLE
rec: Upgrade regression tests to use pytest instead of nose

### PR DESCRIPTION
**Pull request by omoerbeek**
One mysterious failure on Debian bookworm spotted: test_EDNS.py does not seem to set the right edns version on the outgoing query. To be investigated.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [X] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
Reproducer:
```
import enum

class EDNSFlag(enum.IntFlag):
    #: DNSSEC answer OK
    DO = 0x8000

x = 0xffffffff

print(x)
x &= ~EDNSFlag.DO
print(x)
```

on a python 3.8 system:
```
4294967295
EDNSFlag.2147483648|1073741824|536870912|268435456|134217728|67108864|33554432|16777216|8388608|4194304|2097152|1048576|524288|262144|131072|65536|16384|8192|4096|2048|1024|512|256|128|64|32|16|8|4|2|1
```

On a python 3.11 system:
```
4294967295
32767
```

**Comment by peterthomassen** [comment]
There seem to be differences in flag handling between Python 3.11.3 and 3.11.4 (https://github.com/python/cpython/issues/105497). Which 3.11 minor version are you testing on?

**Comment by omoerbeek** [comment]
> There seem to be differences in flag handling between Python 3.11.3 and 3.11.4 ([python/cpython#105497](https://github.com/python/cpython/issues/105497)). Which 3.11 minor version are you testing on?

3.11.2  (also edited above)

**Comment by omoerbeek** [comment]
Output on a python 3.11.4 system:
```
4294967295
4294934527
```
`4294934527 = 0xffff7fff`, which looks good.

**Comment by zeha** [comment]
doesn't seem quite right:

```
  - generated xml file: /__w/pdns/pdns/regression-tests.recursor-dnssec/pytest.xml -
  ================= 441 passed, 6 warnings in 490.86s (0:08:10) ==================
  + ./printlogs.py
  Traceback (most recent call last):
    File "/__w/pdns/pdns/regression-tests.recursor-dnssec/./printlogs.py", line 8, in <module>
      e = xml.etree.ElementTree.parse('nosetests.xml')
    File "/usr/lib/python3.9/xml/etree/ElementTree.py", line 1229, in parse
      tree.parse(source, parser)
    File "/usr/lib/python3.9/xml/etree/ElementTree.py", line 569, in parse
      source = open(source, "rb")
  FileNotFoundError: [Errno 2] No such file or directory: 'nosetests.xml'
```

**Comment by omoerbeek** [comment]
>   FileNotFoundError: [Errno 2] No such file or directory: 'nosetests.xml'

Fixed, thanks!

**Comment by omoerbeek** [comment]
Added code that skip the tests on specific python versions:
```
if sys.version_info >= (3, 11) and sys.version_info <= (3, 11, 3):
            raise SkipTest("Test skipped, see https://github.com/PowerDNS/pdns/pull/12912")
```

File: regression-tests.recursor-dnssec/test_RecDnstap.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def FrameStreamUnixListener(cls, conn, param):
                 fstrm_handle_bidir_connection(conn, lambda data: \
                 param.queue.put(data, True, timeout=2.0))
             except socket.error as e:
-                if e.errno == 9:
+                if e.errno == errno.EBADF or e.errno == errno.EPIPE:
```
Comment (zeha) [inline regression-tests.recursor-dnssec/test_RecDnstap.py:189]: minor style suggestion - feel free to ignore:
```suggestion
                if e.errno in (errno.EBADF, errno.EPIPE):
```

**Review by rgacogne — APPROVED** [review]

**Comment by rgacogne** [comment]
Bad news, this seems broken on 3.11.5 again:
```
$ python badvers.py
4294967295
32767
```

**Pull request merged**